app/lib/three_scale/redis_config.rb

```diff
@@ -15,11 +16,7 @@ def initialize(redis_config = {})
     attr_reader :config
 
     def db
-      value = config.db.presence
-      return value.to_i if value
-      url = config.url.presence
-      return unless url
-      URI.parse(url).path[1..-1].to_s.to_i
+      config.db.to_i
```

Comment:
How about:
1. `raw_config[:db] ||= URI.parse(raw_config[:url].to_s).path[1..].to_i`
and
2. delegate `#db` to `@config`?

Reply:
That would always set DB explicitly to `0` when not provided. That would probably work but I thought it was better to leave it `nil` in that scenario because that's what it did before.

Reply:
I added a unit test: https://github.com/3scale/porta/pull/3800/commits/acd00125a39483b0c854233004164421841c3e8d

Reply:
I think we need to prevent issuing db command when db was not provided because some implementations don't support that. Dunno whether they fail or just do nothing. I think Redis in AWS is one of those not supporting. Probably safer to leave it as original and do not issue additional commands.

Moreover I find it very confusing if incorrect configuration is turned into correct one but possibly not what user expects.

Reply:
> That would always set DB explicitly to `0` when not provided. That would probably work but I thought it was better to leave it `nil` in that scenario because that's what it did before.

I see, OK, makes sense.

Out of curiosity, I checked, and this gem considers value `0` as "no value": https://github.com/redis-rb/redis-client/blob/72a1607ea0fa48725c5198654b2a508120de2ec9/lib/redis_client/config.rb#L168-L170

But I am not sure this is actually the logic that is executed, I didn't check :grimacing: 
The current implementation is fine.

Reply:
@akostadinov your suggestion is equivalent to removing the `#db` method because there's a `#method_missing` method which would delegate `#db` to `config`:

https://github.com/3scale/porta/blob/4a2d49830c01bf9afd67399e5410ee50f9f027e9/app/lib/three_scale/redis_config.rb#L33-L38

On the other hand, the suggestion would be a breaking change because the `#db` method returns `0` now. See the test:

https://github.com/3scale/porta/blob/7a35601253e2a4c647a2dfcfd0441219441e63a6/test/unit/three_scale/redis_config_test.rb#L7-L11

However, you are right when you say `SELECT` is not supported in production. Apparently twemproxy doesn't support it: https://github.com/twitter/twemproxy/blob/master/notes/redis.md#connection

My guess is nobody is using the `#db` method. I couldn't find any call to it in the code base, and apparently the only two places where it was ever being called were removed in https://github.com/3scale/porta/commit/7a78f2a4df5848749e5865c7755f15c2e1fe2605.

So I bet for removing it and delegate it to `config`

Reply:
> Out of curiosity, I checked, and this gem considers value 0 as "no value": https://github.com/redis-rb/redis-client/blob/72a1607ea0fa48725c5198654b2a508120de2ec9/lib/redis_client/config.rb#L168-L170

Good to know

Reply:
I agree with @jlledom, let's remove `#db`.

Reply:
Done in https://github.com/3scale/porta/pull/3800/commits/23bdb35bc9a0da6f9177474433c391c4b404f922